### PR TITLE
CAKE-5314 | Add preferred index

### PR DESCRIPTION
**Pull request by JoshuaRogan**
## Links

* http://wikia-inc.atlassian.net/browse/CAKE-5314

## Description

Use preferred index = 0 for all small units. 

## Reviewers
@Wikia/cake

### CODE REVIEW

File: app/components/post-search-results.js [hunk 1]
```diff
@@ -11,13 +11,17 @@ import extend from '../utils/extend';
...
     return 0;
   }
 
-  const preferredIndex = smallAffiliateUnit.preferredIndex || DEFAULT_AFFILIATE_SLOT;
+  if (smallAffiliateUnit.preferredIndex !== undefined) {
+    preferredIndex = smallAffiliateUnit.preferredIndex;
+  }
 
-  if (preferredIndex < posts.length) {
+  if (preferredIndex > posts.length) {
```
Comment (mklucsarits) [inline app/components/post-search-results.js:24]: ;)

**Review by mklucsarits — APPROVED** [review]
LGTM, although I'm wondering if we should just make 0 the new "default" index instead of even editing the JSON. 🤷‍♂

File: app/components/post-search-results.js (hunk 1 above)
Comment (JoshuaRogan) [reply app/components/post-search-results.js:24]: No one to blame but myself for this error. 🤦‍♂

**Comment by JoshuaRogan** [comment]
> LGTM, although I'm wondering if we should just make 0 the new "default" index instead of even editing the JSON. 🤷‍♂

Good suggestion https://github.com/Wikia/mobile-wiki/pull/1930/commits/18d9d8608696f3564357c543186810f94461aa7b

**Review by mklucsarits — APPROVED** [review]

**Pull request merged**